src/libos/src/fs/inode_file.rs

```diff
@@ -113,6 +113,46 @@ impl File for INodeFile {
         Ok(*offset as i64)
     }
 
+    fn preadv(&self, bufs: &mut [&mut [u8]], offset: usize) -> Result<usize> {
+        if !self.access_mode.readable() {
+            return_errno!(EBADF, "File not readable");
+        }
+        self.offset.lock().unwrap();
```

Comment:
The lock here seems to be useless because the preadv does not change the seek offset of the opened file.